tests/test_interpolation.py

```diff
@@ -462,25 +468,61 @@ def test_resolver_cache_3_dict_list(restore_resolvers: Any) -> None:
 
 
 def test_resolver_no_cache(restore_resolvers: Any) -> None:
-    OmegaConf.register_resolver(
-        "random", lambda _: random.uniform(0, 1), use_cache=False
+    OmegaConf.new_register_resolver(
+        "random", lambda _, __: random.uniform(0, 1), use_cache=False
     )
-    c = OmegaConf.create(dict(k="${random:_}"))
+    c = OmegaConf.create(dict(k="${random:__}"))
     assert c.k != c.k
 
 
 def test_resolver_dot_start(restore_resolvers: Any) -> None:
     """
     Regression test for #373
     """
-    OmegaConf.register_resolver("identity", lambda x: x)
+    OmegaConf.new_register_resolver("identity", lambda _, x: x)
+    c = OmegaConf.create(
+        {"foo_nodot": "${identity:bar}", "foo_dot": "${identity:.bar}"}
+    )
+    assert c.foo_nodot == "bar"
+    assert c.foo_dot == ".bar"
```

Comment:
can you add explicit testing for relative interpolation with resolvers?

```python
add_10(_, v):
  return v + 10

# register_new_resolver(add_10)
```
```yaml
node:
  bar: 10
  foo : ${add_10:${.bar}}  # 20
```

Reply:
Done in c7573c6